### PR TITLE
Updated submodule install docs

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Added documentation of each of the optional dependencies of Tabular
- Added documentation of `autogluon.mxnet`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1091-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1091/1/index.html

File: docs/install-include.rst [hunk 1]
```diff
@@ -164,8 +164,13 @@ Select your preferences below and run the corresponding install commands:
 AutoGluon is modularized into `sub-modules <https://packaging.python.org/guides/packaging-namespace-packages/>`_ specialized for tabular, text, or image data. You can reduce the number of dependencies required by solely installing a specific sub-module via:  `python3 -m pip install <submodule>`, where `<submodule>` may be one of the following options:
 
 - `autogluon.tabular` - only functionality for tabular data (TabularPredictor)
+    - The default installation of `autogluon.tabular` standalone is a skeleton installation.
+    - Install via `pip install autogluon.tabular[all]` to get the same installation as via `pip install autogluon`
+    - Available optional dependencies: `lightgbm,catboost,xgboost,fastai`. These are included in `all`.
```
Comment (jwmueller) [inline docs/install-include.rst:169]: May be good to provide example for novices like:

To run autogluon.tabular with only the optional LightGBM and CatBoost models for example, you can do:
`pip install autogluon.tabular[lightgbm,catboost]`

File: docs/install-include.rst [hunk 2]
```diff
@@ -164,8 +164,13 @@ Select your preferences below and run the corresponding install commands:
...
+    - Experimental optional dependency: `skex`. This will speedup KNN models by 25x in training and inference on CPU. Use `pip install autogluon.tabular[all,skex]` to enable, or `pip install "scikit-learn-intelex<2021.3"` after a standard installation of AutoGluon.
 - `autogluon.vision` - only functionality for computer vision (ImagePredictor, ObjectDetector)
 - `autogluon.text` - only functionality for natural language processing (TextPredictor)
 - `autogluon.core` - only core functionality (Searcher/Scheduler) useful for hyperparameter tuning of arbitrary code/models.
 - `autogluon.features` - only functionality for feature generation / feature preprocessing pipelines (primarily related to Tabular data).
 - `autogluon.extra` - miscellaneous extra functionality such as `Efficient Neural Architecture Search <tutorials/nas/enas_proxylessnas.html>`_
+- `autogluon.mxnet` - miscellaneous extra functionality for MXNet.
```
Comment (jwmueller) [inline docs/install-include.rst:176]: "functionality for MXNet" -> "MXNet functionality"

Comment (jwmueller) [reply docs/install-include.rst:176]: might be helpful to also show how to pip install a submodule from source

Comment (Innixma) [reply docs/install-include.rst]: added

File: docs/install-include.rst (hunk 1 above)
Comment (Innixma) [reply docs/install-include.rst:169]: added

**Comment by szha** [comment]
Job PR-1091-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1091/2/index.html

**Review by jwmueller — APPROVED** [review]
LGTM

**Pull request merged**